src/TemplateConverter.ump

```diff
@@ -0,0 +1,453 @@
+        String rest = bufferedReader.readLine();
+        
+        directive = new DirectiveParser(rest);
+        if (!directive.getIsValid()) {
+            return false;
+        }
+        
+        if (isMainTemplate) {
+            skeletonGenerator = new SkeletonGenerator(directory, directive.getSkeleton());
+        }
+        
+        rest = rest.substring(rest.indexOf("%>") + "%>".length());
+        
+        // If the first line is just for the jet directive, and the next line starts with a code or directive block
+           // then not add a new line
+        // Also, remove any extra spaces
+        if (rest.equals("")) {
+            rest = bufferedReader.readLine();
+            
+            // Should handle this a different way, maybe from a return
+            if (rest == null) {
+                return true;
+            }
+            
+            boolean startsWithCodeOrDirective = false;
+            Operation operation = operationStartingOnLine(rest);
+            int pos = rest.indexOf("<%");
+            
+            if (operation == Operation.Code || operation == Operation.Directive) {
+                startsWithCodeOrDirective = containsOnlySpaces(rest.substring(0, pos));
+            }
+            
+            if (startsWithCodeOrDirective) {
+                rest = rest.substring(pos);
+            } else {
+                elements.add(new BasicElement(NewLine));
+            }
+        }
+    
+        Operation previousOp = Operation.None;
+        
+        // Rest is for the rest of the file
+        while (rest != null) {
+            Operation currentOp = getStartingOperation(rest);
+            switch (currentOp) {
+                case Expression:
+                    rest = parseExpressionBlock(bufferedReader, rest);
+                    break;
+                    
+                case Directive:
+                    // parseDirective -> Right now, just handle 'include file="<template>"
+                    rest = parseDirectiveBlock(bufferedReader, rest);
+                    break;
+                    
+                case Code:
+                    rest = parseCodeBlock(bufferedReader, rest);
+                    break;
+                    
+                default:
+                    rest = parsePureText(bufferedReader, rest, previousOp);
+                    break;
+            }
+            
+            previousOp = currentOp;
+        }
+        
+        if (containsTabs) {
+            printWarning("Contains tabs, which will be converted to 4 spaces");
+        }
+        
+        return true;
+    }
+    
+    private String parseExpressionBlock(BufferedReader bufferedReader, String restOfLine) throws IOException {
+        StringBuffer buffer = new StringBuffer();
+        buffer.append("<<=");
+        
+           String line = restOfLine.substring("<%=".length());
+        
+        // Read all full lines first
+        while (!operationIsFinished(line, "%>")) {
```

Comment:
This end tag feels like a "magic number," it should probably be a static field. There's a couple others that should be done, too.